api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -516,62 +518,125 @@ public function createRecordedEntryAction($entryId, $mediaServerIndex, $liveEntr
-					$maxAppendTimeReached = ($recordedEntryCreationTime + self::SEVEN_DAYS_IN_SECONDS) < time();
-
-					KalturaLog::debug("isNewSession [$isNewSession] getLastBroadcastEndTime [{$dbLiveEntry->getLastBroadcastEndTime()}] getCurrentBroadcastStartTime [{$dbLiveEntry->getCurrentBroadcastStartTime()}]");
-					KalturaLog::debug("recordedEntryCreationTime [$recordedEntryNotYetCreatedForCurrentSession] recordedEntryCreationTime [$recordedEntryCreationTime] getCurrentBroadcastStartTime [{$dbLiveEntry->getCurrentBroadcastStartTime()}]");
-					KalturaLog::debug("maxAppendTimeReached [$maxAppendTimeReached] recordedEntryCreationTime [$recordedEntryCreationTime]");
-
-					if ($dbLiveEntry->getRecordStatus() == RecordStatus::PER_SESSION && $isNewSession && $recordedEntryNotYetCreatedForCurrentSession)
-					{
-						$createRecordedEntry = true;
-					}
-
-					if($dbLiveEntry->getRecordStatus() == RecordStatus::APPENDED && $dbRecordedEntry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE && $maxAppendTimeReached)
-					{
-						$createRecordedEntry = true;
-						$dbLiveEntry->setRecordedEntryId(null);
-						$dbLiveEntry->save();
-					}
-				}
-			}
-			if ($createRecordedEntry)
-			{
-				KalturaLog::info("Creating a recorded entry for ".$dbLiveEntry->getId());
-				$this->createRecordedEntry($dbLiveEntry, $mediaServerIndex);
-			}
+		KalturaLog::info("Checking if recorded entry needs to be created for entry " . $dbLiveEntry->getId());
+		if (!$dbLiveEntry->getRecordStatus() || $dbLiveEntry->getRecordStatus() === RecordStatus::DISABLED)
+		{
+			return false;
+		}
+		if (!in_array($liveEntryStatus, array(EntryServerNodeStatus::BROADCASTING, EntryServerNodeStatus::PLAYABLE)))
+		{
+			return false;
+		}
+		if ($forcePrimaryValidation && $mediaServerIndex !== EntryServerNodeType::LIVE_PRIMARY)
+		{
+			return false;
+		}
+
+		if (!$dbLiveEntry->getRecordedEntryId())
+		{
+			return true;
+		}
+
+		$dbRecordedEntry = entryPeer::retrieveByPK($dbLiveEntry->getRecordedEntryId());
+		if (!$dbRecordedEntry)
+		{
+			return true;
+		}
+
+		if ($dbLiveEntry->getRecordStatus() == RecordStatus::PER_SESSION)
+		{
+			return $this->shouldCreateRecordedEntryForPerSession($dbRecordedEntry, $dbLiveEntry);
+		}
+
+		if ($dbLiveEntry->getRecordStatus() == RecordStatus::APPENDED)
+		{
+			return $this->shouldCreateRecordedEntryForAppend($dbRecordedEntry);
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checking whether should create recorded entry for the "append" record status case
+	 * @param entry $dbRecordedEntry
+	 * @return bool
+	 * @throws Exception
+	 */
+	protected function shouldCreateRecordedEntryForAppend(entry $dbRecordedEntry)
+	{
+		$recordedEntryCreationTime = $dbRecordedEntry->getCreatedAt(null);
+		$isKalturaRecordedLive = $dbRecordedEntry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE; // probably can be removed
```

Comment:
as we suspect we can remove it - lets print here something if it is false - so later we can scan the log and see if we right and it actually can be remove

Reply:
done